HousingManagementSystemApi.Tests/UseCasesTests/RetrieveAddressesUseCaseTests.cs

```diff
@@ -32,15 +41,33 @@ public async Task GivenAPostcode_WhenExecute_GatewayReceivesCorrectInput()
         }
 
         [Fact]
-        public async Task GivenAPostcode_WhenAnAddressExists_GatewayReturnsCorrectData()
+        public async Task GivenAPostcode_WhenAnAddressExistsWithAnEligibleAssetType_ThenThePropertyIsReturned()
         {
             const string TestPostcode = "postcode";
             retrieveAddressesGateway.Setup(x => x.SearchByPostcode(TestPostcode))
-                .ReturnsAsync(new PropertyAddress[] { new() { PostalCode = TestPostcode } });
+                .ReturnsAsync(new PropertyAddress[] { new() { PostalCode = TestPostcode, Reference = new Reference { ID = "assetId" } } });
+
+            retrieveAssetGateway.Setup(x => x.RetrieveAsset("assetId"))
+                .ReturnsAsync(new AssetResponseObject { AssetType = AssetType.Dwelling });
+
             var result = await retrieveAddressesUseCase.Execute(TestPostcode);
             result.First().PostalCode.Should().Be(TestPostcode);
         }
 
+        [Fact]
+        public async Task GivenAPostcode_WhenAnAddressExistsWithAIneligibleAssetType_ThenAPropertyIsNotReturned()
```

Comment:
Please rename to be grammatically correct (A -> An), i.e.
```suggestion
        public async Task GivenAPostcode_WhenAnAddressExistsWithAnIneligibleAssetType_ThenAPropertyIsNotReturned()
```